### PR TITLE
feat: expose Wallet::policies method

### PR DESCRIPTION
**Pull request by thunderbiscuit**
I don't know wtf is going on with this branch/PR and the naming but it's really not liking me pushing to it.

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
@reez this is the same PR as #612 and #626, but rebased.

File: bdk-ffi/src/types.rs [hunk 1]
```diff
@@ -237,3 +245,255 @@ pub struct KeychainAndIndex {
...
+    AbsoluteTimelock {
+        value: LockTime,
+    },
+    RelativeTimelock {
+        value: u32,
+    },
+
+    Multisig {
+        keys: Vec<PkOrF>,
+
```
Comment (reez) [inline bdk-ffi/src/types.rs:317]: nit: remove line

File: bdk-ffi/src/types.rs [hunk 2]
```diff
@@ -237,3 +245,255 @@ pub struct KeychainAndIndex {
...
+        threshold: u64,
+    },
+
+    Thresh {
+        items: Vec<Arc<Policy>>,
+
```
Comment (reez) [inline bdk-ffi/src/types.rs:323]: nit: remove line

File: bdk-ffi/src/types.rs [hunk 3]
```diff
@@ -237,3 +245,255 @@ pub struct KeychainAndIndex {
...
+        n: u64,
+        m: u64,
+        items: Vec<u64>,
+        sorted: Option<bool>,
+        conditions: HashMap<Vec<u32>, Vec<Condition>>,
+    },
+    Complete {
+        condition: Condition,
+    },
+
```
Comment (reez) [inline bdk-ffi/src/types.rs:427]: nit: remove line

File: bdk-ffi/src/types.rs [hunk 4]
```diff
@@ -237,3 +245,255 @@ pub struct KeychainAndIndex {
...
+                n: n as u64,
+                m: m as u64,
+                items: items.iter().map(|e| e.to_owned() as u64).collect(),
+                sorted,
+                conditions: conditions
+                    .into_iter()
+                    .map(|(index, conditions)| {
+                        (
+                            index.iter().map(|e| e.to_owned() as u32).collect(),
+                            conditions.into_iter().map(|e| e.into()).collect(), // Convert each `Condition` to `YourType`
```
Comment (reez) [inline bdk-ffi/src/types.rs:472]: nit: remove code comment potentially?

File: bdk-ffi/src/types.rs [hunk 5]
```diff
@@ -237,3 +245,255 @@ pub struct KeychainAndIndex {
...
+    Hash160Preimage {
+        hash: String,
+    },
+    AbsoluteTimelock {
+        value: LockTime,
+    },
+    RelativeTimelock {
+        value: u32,
+    },
+
```
Comment (reez) [inline bdk-ffi/src/types.rs:314]: nit: remove line

**Review by reez — APPROVED** [review]

**Comment by thunderbiscuit** [comment]
Thanks! Indeed I found a few places where newlines where missing and a few places where newlines where not applied systematically. Fixed all that now.

**Pull request merged**